WordPress/src/main/java/org/wordpress/android/ui/reader/utils/FeaturedImageUtils.kt

```diff
@@ -0,0 +1,35 @@
+package org.wordpress.android.ui.reader.utils
+
+import java.net.MalformedURLException
+import java.net.URL
+import javax.inject.Inject
+
+class FeaturedImageUtils
+@Inject constructor() {
+    fun showFeaturedImage(
+        featuredImage: String,
+        postText: String
+    ): Boolean {
+        return try {
+            val featuredImageUrl = URL(featuredImage)
+            val endIndex = featuredImageUrl.path.lastIndexOf("-", featuredImageUrl.path.lastIndexOf("/"))
+            val featuredImageFile = featuredImageUrl.file
+            val featuredImagePath = if (endIndex > 0 && !featuredImageFile.startsWith("-")) {
+                featuredImageUrl.path.substring(0, endIndex)
+            } else {
+                featuredImageUrl.path
+            }
+            when {
+                postText.contains(featuredImage) -> false
+                postText.contains(featuredImagePath) && featuredImagePath != featuredImageFile -> false
+                else -> true
+            }
+        } catch (e: MalformedURLException) {
+            false
+        }
```

Comment:
Not fully sure to get al the nuances of the logic of this function and maybe it's a bit too much defensive coding but I want to mention for your consideration 😊. I was wondering if it can make sense here to widen the exception so that we in general default to false and possibly adding a log. Just thinking aloud, let me know wdyt.

Reply:
the idea here is that if the featured image URL is malformed, we don't want to show the featured image (because we can't anyway 😄 ). I can add a log here though.